cmd/juju/charmhub/download.go

```diff
@@ -244,14 +247,25 @@ func (c *downloadCommand) Run(cmdContext *cmd.Context) error {
 		return errors.Trace(err)
 	}
 	channelMap = filterInfoChannelMap(channelMap, filterFn)
-	revision, found := locateRevisionByChannel(c.sortInfoChannelMap(channelMap), charmChannel)
+	revisions, found := locateRevisionByChannel(c.sortInfoChannelMap(channelMap), charmChannel)
 	if !found {
 		if c.series != "" {
 			return errors.Errorf("%s %q not found for %q channel matching %q series", info.Type, c.charmOrBundle, c.channel, c.series)
 		}
 		return errors.Errorf("%s %q not found with in the channel %q", info.Type, c.charmOrBundle, c.channel)
 	}
 
+	if selectAll {
+		// Ensure we have just one architecture.
+		if !hasMatchingArch(revisions) {
+			return errors.Errorf("ambiguous architectures found, use --arch flag to specify architecture")
```

Comment:
Can you include the list of found arches in the error?

Reply:
Also perhaps rephrase the error as `multiple architectures (%s) found, use --arch flag to specify the one to download`